### PR TITLE
feat(openchallenges): initialize basic docs site

### LINKED ISSUE #2395: [Feat] Create documentation site
Currently, all of our docs are accessible from the repo only.  It would be nice if we could host a "pretty" view of the docs as well.

I plan to follow the [Diátaxis](https://diataxis.fr/) structure when porting over the docs.

### PR DESCRIPTION
**Pull request by vpchung**
Fixes #2395 

## Changelog
- add [MkDocs](https://www.mkdocs.org/) and [Material for MkDocs](https://squidfunk.github.io/mkdocs-material/) configurations
- "port" over some docs into the docs site
- add CI workflow for building docs

> [!WARNING]
> There's a lot going on in our current `docs` folder that may no longer be relevant and/or too complex to belong on the docs site.  I highly suggest we revisit the docs in another PR to re-organize and/or update the contents

## Preview

docs

## Local development

To test out this branch, you will need to:

1. install mkdocs, mkdocs-material, and termynal, ideally in a virtual environment (I used conda):

```
pip install mkdocs mkdocs-material termynal
```

2. run mkdocs serve:

```
mkdocs serve
```

This will open the docs at `http://127.0.0.1:8000/`

> [!NOTE]
> Because it uses the `8000` port, you should not be running the web app while testing out this feature.

### CODE REVIEW

**Review by gaiaandreoletti — APPROVED** [review]

File: .github/workflows/build-docs.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: Build Docs
+on:
+  push:
```
Comment (tschaffter) [inline .github/workflows/build-docs.yml:3]: Run only when [specific files are changed](https://github.com/Sage-Bionetworks/sage-monorepo/blob/333301ed6c3db53574a02989ce393b7fd258ce1c/.github/workflows/build-devcontainer-image.yml#L6-L9) to save compute resources?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,163 +1 @@
...
-
-<!-- Links -->
-
-[Code of Conduct]: CODE-OF-CONDUCT.md
-[upstream remote]: https://help.github.com/en/articles/configuring-a-remote-for-a-fork
-[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
-[Forking Workflow]: https://www.atlassian.com/git/tutorials/comparing-workflows/forking-workflow
-[package.json]: ../package.json
-[README.md]: ../README.md
+Please read the Contributing guidelines on the docs site.
```
Comment (tschaffter) [inline CONTRIBUTING.md:1]: Add URL

File: .github/workflows/build-docs.yml (hunk 1 above)
Comment (vpchung) [reply .github/workflows/build-docs.yml:3]: Good idea

File: CONTRIBUTING.md (hunk 2 above)
Comment (vpchung) [reply CONTRIBUTING.md:1]: I was planning on adding the URL once we finalize the navigation links.

File: .github/workflows/build-docs.yml (hunk 1 above)
Comment (vpchung) [reply .github/workflows/build-docs.yml:3]: I specified that the CI should only trigger if there are changes to the mkdocs config file, any of the markdown files in `docs/` or any file in `docs/images/`

**Comment by tschaffter** [comment]
It would be nice to prepare, build, serve and maybe publish the docs using the Nx API (e.g. `nx build monorepo-docs`, `nx serve monorepo-docs`) like we do for OC API docs.

File: .github/workflows/build-docs.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Configure Git Credentials
+        run: |
+          git config user.name github-actions[bot]
+          git config user.email 41898282+github-actions[bot]@users.noreply.github.com
```
Comment (tschaffter) [inline .github/workflows/build-docs.yml:19]: Where does this email address come from?

**Review by tschaffter — APPROVED** [review]

File: .github/workflows/build-docs.yml (hunk 3 above)
Comment (vpchung) [reply .github/workflows/build-docs.yml:19]: It's the email for the github actions bot.

**Pull request merged**